File: tables.php
<?php

echo "disabled\n"; exit;

require("include/mysql_connect.php");
mysql_query("DROP TABLE IF EXISTS ".DB_TABLE.";") or die(mysql_error());
mysql_query("CREATE TABLE ".DB_TABLE." (
  id INT NOT NULL AUTO_INCREMENT,
  signature VARCHAR(3) NOT NULL,
  count INT NOT NULL,
  successes INT NOT NULL,
  failures INT NOT NULL,
  duration INT NOT NULL,
  size BIGINT UNSIGNED NOT NULL,
  retries INT NOT NULL,
  date TIMESTAMP default CURRENT_TIMESTAMP NOT NULL,
  PRIMARY KEY(id)
  ) engine=InnoDB AUTO_INCREMENT=1;") or die(mysql_error());

?>

File: tables.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import os
import sqlite3

DB_FILENAME = os.path.join('work', 'stats.db')
DB_TABLE = 'stats'

statement = 'CREATE TABLE %s (\n\
  id INT PRIMARY KEY,\n\
  date TIMESTAMP default CURRENT_TIMESTAMP NOT NULL,\n\
  signature VARCHAR NOT NULL,\n\
  count INT NOT NULL,\n\
  size BIGINT UNSIGNED NOT NULL,\n\
  successes INT NOT NULL,\n\
  failures INT NOT NULL,\n\
  duration INT NOT NULL,\n\
  retries INT NOT NULL\n\
  );' % DB_TABLE

def create_table(directory):
    
    database_file = os.path.join(directory, DB_FILENAME)
    
    try:
        with open(database_file):
            return False
    except IOError: pass
    
    connection = None
    
    try:
    
        connection = sqlite3.connect(database_file)
        cursor = connection.cursor()
        cursor.execute(statement)
        return True
    
    finally:
    
        if connection:
            connection.close()
    
    return False

#if __name__ == '__main__':
#    directory = os.path.abspath(os.path.dirname(__file__)
#    create_table()
